### ISSUE TITLE
Bug in Switch option...current build

### ISSUE THREAD

**Issue by hookedonweb** [evertiro/historical-redux2#43]
I can not set the switch option to 'on' as a default. Even if you remove the extra white space as in the sample, the switch is always in the off position.
![2013-01-08_1052](https://f.cloud.github.com/assets/3218655/51206/11c996fe-59c5-11e2-9d2a-45c59253e261.png)
![2013-01-08_1053](https://f.cloud.github.com/assets/3218655/51205/0c360984-59c5-11e2-8cd1-159b0b7414ba.png)

**Comment by pcescato** [comment]
It seems that this issue is the same that issue 41 I've opened. In fact, after removing some validated fields, I've no more problems, neither with checkboxes, neither with colors... perhaps some buggy code in a validation function breaks std process, I've not looked at code in fact, too tired and involved in my theme building.

**Comment by denis0706** [comment]
i have tried to remove all validated field but it didn't help.
At first sight it seems that there is a problem in defaults.php
Function _set_default_options (line:155) checks if redux option array exists in wp DB, 
but doesn't check if there is option name in redux option array keys.

**Comment by psaikali** [comment]
Does this merge fix this checkbox bug for you? 
If my checkbox (switch => true or false) has std => '1' (or 1), unchecking the checkbox doesn't get saved, it's always checked...

**Comment by denis0706** [comment]
Sorry, i have hurried.
The problem with checkbox is that if it is uncheked its value are not sent.
So we need to check it:

``` php
if(isset($field['type']) && $field['type']=='checkbox'){
    if(!isset($plugin_options[$field['id']]))
    $plugin_options[$field['id']] = 0;
}
```

Add this code to defaults.php at **function _validate_values** after **$field['section_id'] = $k;**  _(line:498)_

**Comment by psaikali** [comment]
Awesome, I'll try that soon, thx a lot!

**Comment by denis0706** [comment]
also we should check other checkbox fields type: _checkbox_hide_all_, _checkbox_hide_below_
I have found another problem with switch style checkbox.
Sometimes after  press "Save changes" button, checkbox stop working. Javascript catches change and fire warning "Settings have changed", but checkbox state  isn't changed. It's look like switch "label" loose relation checkbox input, either we use `<label><input/></label>` or we use  `<input/><label for="input_id"></label>`
it doesn't happen after option page reload, but after 3-5 times pressing "save button" there is this problem.

**Comment by denis0706** [comment]
Solved. The problem with switch style exists only in opera, caused by display:none; for input.
So we should hide checkbox not using display:none
We could set its position to absolute and set opacity:0 or huge offset for browser with no opacity support (left:-9999px;)

**Comment by evertiro** [comment]
This bug has been resolved.

**Comment by hookedonweb** [comment]
has the fix been applied to a the existing download build?

**Comment by evertiro** [comment]
It was fixed in https://github.com/ghost1227/Redux-Framework/commit/0ed37d096eb2794eb0d12fa0530998e396818e52

**Comment by hookedonweb** [comment]
How does one know if they are using the right build if the version number is not incremented? Pre-bug report version was 1.0.1 and post bug fix is version 1.0.1. Is there a patch file to download somewhere I am not seeing?

**Comment by evertiro** [comment]
If you're using the development builds, the best practice is probably to use Redux as a submodule. Instructions for doing this are in the readme file. Beyond that, just keep an eye on the commit logs for changes that affect you!
